Sources/AsyncKit/ConnectionPool/ConnectionPoolError.swift

```diff
@@ -2,4 +2,6 @@
 public enum ConnectionPoolError: Error {
     /// The connection pool has shutdown.
     case shutdown
+    /// The connection creationg timed out.
+    case connectionCreateTimeout
```

Comment:
We can't add new cases to this enum unfortunately. I'd recommend a new struct or enum.